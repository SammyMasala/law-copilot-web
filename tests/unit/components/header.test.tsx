import React, { createContext, useState } from "react";
import "@testing-library/jest-dom"
import {render, screen, cleanup} from "@testing-library/react"
import Header from "../../../src/components/header/header";

afterEach(()=> {
    cleanup();
})

describe("Header Component", () => {
    const mockContext:React.Context<any> = createContext(null)
    const MockContextProvider = ({children}: any) => {
        const [sessionURL, setSessionURL] = useState<string>("https://www.test.com")
        return (
            <mockContext.Provider 
                value={{sessionURL}}>
                    {children}
            </mockContext.Provider>
        )
    }
    
    test("initial state", async () => {
        const TestComponent:React.FC = () => {
            return (
                <MockContextProvider>
                    <Header context={mockContext}/>
                </MockContextProvider>
            )
        }
        render(<TestComponent />)
        const title = screen.getByText("Law Copilot")
        expect(title).toBeInTheDocument();
    })

    test("snapshot", () => {   
        const TestComponent:React.FC = () => {
            return (
                <MockContextProvider>
                    <Header context={mockContext}/>
                </MockContextProvider>
            )
        }     
        const {asFragment} = render(<TestComponent />)
        expect(asFragment()).toMatchSnapshot();
    })
})