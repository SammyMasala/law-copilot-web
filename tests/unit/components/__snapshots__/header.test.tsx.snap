// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-fluid"
  >
    <div
      class="d-flex row"
    >
      <div
        class="d-flex justify-content-start col"
      >
        <img
          class="bg-light rounded-circle"
          src=""
        />
        <div>
           Law Copilot 
        </div>
      </div>
      <div
        class="d-flex justify-content-center col-2"
      >
        <a
          href="https://github.com/SammyMasala/law-copilot-web"
        >
          <img
            class="bg-light rounded-circle"
            src=""
          />
        </a>
      </div>
      <div
        class="d-flex justify-content-end col"
      >
        <div>
          Icons by:
        </div>
        <a
          href="https://icons8.com/icons"
        >
          Icon8
        </a>
      </div>
    </div>
    <div
      class="d-flex row"
    >
      <div
        class="d-flex justify-content-center col-12"
      >
        *** SAVE THIS LINK TO YOUR SESSION ***
        <a
          href="https://www.test.com"
        >
          https://www.test.com
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
